modules/mobile/app/models/mobile/v0/adapters/immunizations.rb

```diff
@@ -9,16 +9,17 @@ def parse(immunizations)
             immunization = i[:resource]
             vaccine_code = immunization[:vaccine_code]
             cvx_code = vaccine_code[:coding].first[:code].to_i
+            vaccine = Mobile::V0::Vaccine.find_by(cvx_code: cvx_code)
```

Comment:
This is happening inside a map so it's a SQL call for each iteration. Maybe there's a way to extract the CVX codes we need to find first, make one ActiveRecord find call and then update the model from that list? It would mean mapping over the date twice but I think that would be more performant than multiple SQL calls.

Reply:
I'll look into that.